packages/ui/src/components/avatar/avatar.web.tsx

```diff
@@ -1,114 +1,155 @@
-import { forwardRef } from 'react';
+import {
+  ComponentPropsWithoutRef,
+  ElementRef,
+  ReactElement,
+  cloneElement,
+  forwardRef,
+} from 'react';
 
 import * as RadixAvatar from '@radix-ui/react-avatar';
-import { type RecipeVariantProps, css, cva } from 'leather-styles/css';
-import { type HTMLStyledProps, styled } from 'leather-styles/jsx';
+import { styled } from 'leather-styles/jsx';
 
-export const defaultFallbackDelay = 600;
-export function getAvatarFallback(val: string) {
-  return val.slice(0, 2);
+import { isDefined } from '@leather.io/utils';
+
+import { IconProps } from '../../icons/icon/create-icon.web';
+import { AvatarSize, AvatarVariant, defaultFallbackDelay, iconSizeMap } from './avatar.shared';
+
+type AvatarElement = ElementRef<typeof AvatarRoot>;
+
+export interface AvatarProps extends ComponentPropsWithoutRef<typeof AvatarRoot> {
+  size?: AvatarSize;
+  variant?: AvatarVariant;
+  icon?: ReactElement;
+  indicator?: ReactElement;
+  image?: string;
+  imageAlt?: string;
+  fallback?: string;
+  fallbackDelayMs?: number;
 }
 
-const avatarRecipe = cva({
+export const Avatar = forwardRef<AvatarElement, AvatarProps>((props, ref) => {
+  const {
+    // TODO: The default size is temporarily set to 'lg' to look correct with current design.
+    //       Set to 'xl', once LEA-2111 is ready.
+    size = 'lg',
+    variant = 'circle',
+    icon,
+    indicator,
+    image,
+    imageAlt,
+    fallback,
+    outlineColor,
+    fallbackDelayMs = defaultFallbackDelay,
+    ...rest
+  } = props;
+  return (
+    <AvatarRoot
+      ref={ref}
+      size={size}
+      variant={variant}
+      outlineColor={
+        outlineColor ??
+        (isDefined(image) ? 'ink.border-transparent' : 'ink.component-background-hover')
+      }
+      {...rest}
+    >
+      <AvatarImage src={image} alt={imageAlt ?? fallback} />
+      <AvatarIcon avatarSize={size} icon={icon} />
+      {fallback ? <AvatarFallback delayMs={fallbackDelayMs}>{fallback}</AvatarFallback> : null}
+      {indicator ? (
+        <styled.div
+          bg="ink.background-primary"
+          borderRadius="round"
+          position="absolute"
+          bottom={-4}
+          right={-4}
+          padding={3}
+        >
+          {indicator}
+        </styled.div>
+      ) : null}
+    </AvatarRoot>
+  );
+});
+
+Avatar.displayName = 'Avatar';
+
+const AvatarRoot = styled(RadixAvatar.Root, {
```

Comment:
What do you think about having a separate `styles` file purely to keep the component size down and easier to work with?

Reply:
I find this kind of separation a bit arbitrary to be honest. Extracting things like styles, helpers, types that only have one consumer into separate files does more harm then good: 
1. More diffs across files for simple changes 
2. Unnecessary exports floating around the app/packages slowing down the compiler, messing with editor auto-import suggestions 
3. Potential circular imports, when sharing dependencies between these modules, which causes to split into even more files deepening the issue.

I feel like excessive file organization is a bit like sweeping under the rug–gives us false sense of decoupling for things that are inherently coupled.

Reply:
OK 👍 I just don't like working with massive files 😄